examples/connext_dds/detect_samples_dropped/c++/drop_publisher.cxx

```diff
@@ -0,0 +1,219 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2020.  All rights reserved.
+* RTI grants Licensee a license to use, modify, compile, and create derivative
+* works of the software solely for use with RTI Connext DDS. Licensee may
+* redistribute copies of the software provided that all such copies are subject
+* to this license. The software is provided "as is", with no warranty of any
+* type, including any warranty for fitness for any purpose. RTI is under no
+* obligation to maintain or support the software. RTI shall not be liable for
+* any incidental or consequential damages arising out of the use or inability
+* to use the software.
+*/
+
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "drop.h"
+#include "dropSupport.h"
+#include "ndds/ndds_cpp.h"
+#include "application.h"
+
+using namespace application;
+
+static int shutdown_participant(
+    DDSDomainParticipant *participant,
+    const char *shutdown_message,
+    int status);
+
+int run_publisher_application(unsigned int domain_id, unsigned int sample_count)
+{
+    // Start communicating in a domain, usually one participant per application
+    DDSDomainParticipant *participant =
+    DDSTheParticipantFactory->create_participant(
+        domain_id,
+        DDS_PARTICIPANT_QOS_DEFAULT,
+        NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (participant == NULL) {
+        return shutdown_participant(participant, "create_participant error", EXIT_FAILURE);
+    }
+
+    // A Publisher allows an application to create one or more DataWriters
+    DDSPublisher *publisher = participant->create_publisher(
+        DDS_PUBLISHER_QOS_DEFAULT,
+        NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (publisher == NULL) {
+        return shutdown_participant(participant, "create_publisher error", EXIT_FAILURE);
+    }
+
+    // Register the datatype to use when creating the Topic
+    const char *type_name = dropTypeSupport::get_type_name();
+    DDS_ReturnCode_t retcode =
+    dropTypeSupport::register_type(participant, type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        return shutdown_participant(participant, "register_type error", EXIT_FAILURE);
+    }
+
+    // Create a Topic with a name and a datatype
+    DDSTopic *topic = participant->create_topic(
+        "Example drop",
+        type_name,
+        DDS_TOPIC_QOS_DEFAULT,
+        NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        return shutdown_participant(participant, "create_topic error", EXIT_FAILURE);
+    }
+
```

Comment:
Add a comment explaining that here begins the changes for the example (and add another one at the end)